Add edits in the locked state

agnieszkajarosikloj opened the pull request:
## Description

This PR adds ability to edit a locked expenditure, after clicking edit icon. The state of the expenditure is revert to an editable. Also adds a modal where user can confirm the changes and displays newly created linked motion. UI only.
[Figma Link - Showing edit icon](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=2416%3A313415)
[Figma Link - Showing edit state](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=2416%3A314175)
[Figma Link - Showing confirm edit modal](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=2416%3A314175)
[Figma Link - Showing linked edit motions](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=2420%3A1074809)

**New stuff** ✨

* Edit button that reverts to an editable state
<img width="232" alt="Screenshot 2022-07-14 at 14 48 48" src="https://user-images.githubusercontent.com/91876137/178986098-0c335107-5ae0-4b06-8880-331011139afd.png">
* Modal where user can confirm changes
<img width="609" alt="Screenshot 2022-07-14 at 15 20 52" src="https://user-images.githubusercontent.com/91876137/178992223-ccf4840e-266d-4631-ba1b-1fab5b73cc0d.png">

**Changes** 🏗

* The Stages component has been changed to separate the Stages inside the form from the one with locked values. It has been done to avoid calling the 'useFormikContext' hook outside of formik context.
* StakeExpenditureDialog has been changed - used only Toggle with correct props
<img width="855" alt="Screenshot 2022-07-27 at 18 41 33" src="https://user-images.githubusercontent.com/91876137/181302881-44429deb-f853-4b90-a693-0b6a865e40db.png">



Resolves #3492

arrenv commented on src/modules/dashboard/components/Dialogs/EditExpenditureDialog/EditExpenditureDialog.tsx:
```diff
@@ -0,0 +1,208 @@
+import EditExpenditureDialogForm from './EditExpenditureDialogForm';
+
+export const MSG = defineMessages({
+  header: {
+    id: 'dashboard.EditConfirmDialog.header',
+    defaultMessage: 'Create a motion to edit payment',
+  },
+  headerForce: {
+    id: 'dashboard.EditConfirmDialog.headerForce',
+    defaultMessage: 'Create a motion to edit expenditure',
```
We can combine these and have them as "Create a motion to edit payment"

arrenv commented on src/modules/dashboard/components/Dialogs/EditExpenditureDialog/EditExpenditureDialog.tsx:
```diff
@@ -0,0 +1,208 @@
+  },
+  confirmText: {
+    id: 'dashboard.EditConfirmDialog.confirmText',
+    defaultMessage: 'Create motion',
```
Can we make it "Create Motion" with the capital "M"

arrenv reviewed (changes requested):
Thank you @agnieszkajarosikloj 

I am getting a unique key error in the console.

![0181a02f-da39-4c41-bd2c-54f5cc9626be](https://user-images.githubusercontent.com/33682027/179282355-81ec9176-1759-4b58-b198-b063b422ccdd.png)

I am able to Update even if no values were changed, I think it would be better to disable the Update button until values are changed. Or, show a value in the dialog that no values have been changed in the same `FormSection` Element

![c202cd25-87c3-45b2-bf62-5a55f5ab9842](https://user-images.githubusercontent.com/33682027/179282360-4301b19a-4262-4fd2-a49a-a6422c83c9c8.png)

The token and the value in the `FormSection` are not the same font size, and is not vertically aligned.

After clicking discard with the last element, you get the same empty dialog issue.

![c887ad57-5c5c-4140-bdc1-782fd32ef3c0](https://user-images.githubusercontent.com/33682027/179282363-a82a47da-bf56-4ef5-a53d-f701b8f4a787.png)


![c202cd25-87c3-45b2-bf62-5a55f5ab9842](https://user-images.githubusercontent.com/33682027/179282360-4301b19a-4262-4fd2-a49a-a6422c83c9c8.png)

willm30 reviewed (changes requested):
@agnieszkajarosikloj  Made a few comments below 🙂

1. Alignment here looks off when changing amount from one coin to two (e.g. not centered vertically).
![new-amount](https://user-images.githubusercontent.com/64402732/179963948-246b0bfd-cb55-485e-bc9d-2fbf2ac9b904.png)

1. When toggling force, I would say the warning should have some bottom padding.
![annotation-padding](https://user-images.githubusercontent.com/64402732/179963977-be200a60-badc-49b1-a8e3-4f30d91aacf3.png)

1. When changing the delay, per the design it would be "1h" which would also remove the issue with the "hours" being plural here. 
![hours](https://user-images.githubusercontent.com/64402732/179964963-13b78a72-4536-45ec-9414-d36e36959e25.png)

1. There's too much space beneath the first row compared with the design, and also, I think the letter spacing of the labels in the design is greater. I believe the colour is also a slightly lighter grey.
![spacing](https://user-images.githubusercontent.com/64402732/179963966-c0457771-4fe5-4ca5-aa59-bd343191e73f.png)

1. I know it's not strictly related to this issue, but "F. ex." should be replaced with "E.g."
![fex](https://user-images.githubusercontent.com/64402732/179963987-67c70606-47a7-42ef-884c-6282aa9c8058.png)
 
1. The label should be in the center of the icon. If you set the \<i\> tag's display to `display: flex` it solves it.
![payments-vertical](https://user-images.githubusercontent.com/64402732/179965715-ae313471-5c0d-424b-9978-24be61272d1e.png)
 
1. The design implies the ability to change owner, however there is no user picker dropdown at the owner section when you click edit. 

![change-owner](https://user-images.githubusercontent.com/64402732/179963795-d67fdc0f-edb2-4853-9a09-b9558ca1fdb1.png)

![owner-picker](https://user-images.githubusercontent.com/64402732/179966364-c2acec7a-4e06-459e-aaa6-5cbe1084973a.png)

willm30 commented:
@agnieszkajarosikloj A question for you: when I edit values and confirm, they aren't updating in the list in the same "dropdown" style as the original. In the screenshots below I added two new recipients but can't see what I added. Is that in scope for this PR?
![edit-locked](https://user-images.githubusercontent.com/64402732/181311144-7b41c9e6-fd56-46b8-935e-351d19c3d4d1.png)

agnieszkajarosikloj commented:
> @agnieszkajarosikloj A question for you: when I edit values and confirm, they aren't updating in the list in the same "dropdown" style as the original. In the screenshots below I added two new recipients but can't see what I added. Is that in scope for this PR? ![edit-locked](https://user-images.githubusercontent.com/64402732/181311144-7b41c9e6-fd56-46b8-935e-351d19c3d4d1.png)

I added a temporary function that needs to be replaced with a call to backend. Now, if you "force" edition, you'll see your changes immediately. When you create a motion to change expenditure, you'll see changes after setTimeout.

willm30 commented:
> I added a temporary function that needs to be replaced with a call to backend. Now, if you "force" edition, you'll see your changes immediately. When you create a motion to change expenditure, you'll see changes after setTimeout.

@agnieszkajarosikloj Ok thanks for clarifying. A couple more notes:

1. When you "force" an update, it tells you an annotation is required but lets you submit the form without one
![force-payment](https://user-images.githubusercontent.com/64402732/181490027-118ca2c8-4cb4-4632-8249-4ca1f5a66b77.png)

2. I have noticed elsewhere in the project the colour of the "bin" icon when deleting a recipient is red but here it's grey. Mentioning just in case.

Approving in advance as is working well for me.

Julianahlk reviewed (changes requested):
Good work so far! A few changes from my end:

1. I noticed the completed button color in this PR is blue which differs from other PRs so for consistency can we keep it either all blue or grey?  
<img width="469" alt="Screen Shot 2022-07-29 at 7 46 36 PM" src="https://user-images.githubusercontent.com/71563622/181817985-ec558560-6677-4421-8577-0d65b2952316.png">

2. In the example below, the button completed should not be showing up since there is a claim delay of 1 hour which doesn't match the actual state or eligibility to claim. 
<img width="1430" alt="Screen Shot 2022-07-29 at 7 48 30 PM" src="https://user-images.githubusercontent.com/71563622/181818483-ca476aa6-43ca-474b-bdfa-90182662e3e2.png">

I was expecting the orange button show up instead and the claim button to be in its off state.
<img width="364" alt="Screen Shot 2022-07-29 at 8 03 58 PM" src="https://user-images.githubusercontent.com/71563622/181818590-255117bd-cec8-4c27-bd62-6a51f8a57937.png">

3. Value is supposed to be aligned to the top or bottom?
<img width="405" alt="Screen Shot 2022-07-29 at 7 57 19 PM" src="https://user-images.githubusercontent.com/71563622/181819282-061b0388-aae1-4d1b-b95d-6476a0a10eae.png">
<img width="359" alt="Screen Shot 2022-07-29 at 7 58 15 PM" src="https://user-images.githubusercontent.com/71563622/181819288-c231238b-3d0f-4720-954a-0baf7e59515e.png">

4. Can we align the following text to the left: "Recipient has been deleted"
<img width="495" alt="Screen Shot 2022-07-29 at 8 13 42 PM" src="https://user-images.githubusercontent.com/71563622/181819978-63829fab-780a-4020-b6ae-a445f09da581.png">
5. Last thing, I noticed this orange dot doesn't have a tooltip 
<img width="503" alt="Screen Shot 2022-07-29 at 8 16 58 PM" src="https://user-images.githubusercontent.com/71563622/181820497-157066ba-16eb-4ffe-82d5-325927b34eab.png">
And the blue dots do 
<img width="459" alt="Screen Shot 2022-07-29 at 8 17 17 PM" src="https://user-images.githubusercontent.com/71563622/181820549-392dedfe-7a41-4d37-bef9-dbca92ad0c8e.png">

agnieszkajarosikloj commented:
> Good work so far! A few changes from my end:
> 
> 1. I noticed the completed button color in this PR is blue which differs from other PRs so for consistency can we keep it either all blue or grey?
> 
> <img alt="Screen Shot 2022-07-29 at 7 46 36 PM" width="469" src="https://user-images.githubusercontent.com/71563622/181817985-ec558560-6677-4421-8577-0d65b2952316.png">
> 
> 2. In the example below, the button completed should not be showing up since there is a claim delay of 1 hour which doesn't match the actual state or eligibility to claim.
> 
> <img alt="Screen Shot 2022-07-29 at 7 48 30 PM" width="1430" src="https://user-images.githubusercontent.com/71563622/181818483-ca476aa6-43ca-474b-bdfa-90182662e3e2.png">
> 
> I was expecting the orange button show up instead and the claim button to be in its off state. <img alt="Screen Shot 2022-07-29 at 8 03 58 PM" width="364" src="https://user-images.githubusercontent.com/71563622/181818590-255117bd-cec8-4c27-bd62-6a51f8a57937.png">
> 
> 3. Value is supposed to be aligned to the top or bottom?
> 
> <img alt="Screen Shot 2022-07-29 at 7 57 19 PM" width="405" src="https://user-images.githubusercontent.com/71563622/181819282-061b0388-aae1-4d1b-b95d-6476a0a10eae.png">
> 
> <img alt="Screen Shot 2022-07-29 at 7 58 15 PM" width="359" src="https://user-images.githubusercontent.com/71563622/181819288-c231238b-3d0f-4720-954a-0baf7e59515e.png">
> 
> 4. Can we align the following text to the left: "Recipient has been deleted"
> 
> <img alt="Screen Shot 2022-07-29 at 8 13 42 PM" width="495" src="https://user-images.githubusercontent.com/71563622/181819978-63829fab-780a-4020-b6ae-a445f09da581.png">
> 
> 5. Last thing, I noticed this orange dot doesn't have a tooltip
> <img alt="Screen Shot 2022-07-29 at 8 16 58 PM" width="503" src="https://user-images.githubusercontent.com/71563622/181820497-157066ba-16eb-4ffe-82d5-325927b34eab.png">
> 
> And the blue dots do
> <img alt="Screen Shot 2022-07-29 at 8 17 17 PM" width="459" src="https://user-images.githubusercontent.com/71563622/181820549-392dedfe-7a41-4d37-bef9-dbca92ad0c8e.png">
> 
> Thank you!

@Julianahlk thank you for comments.
1. I've fixed 'Completed' button color in this PR: https://github.com/JoinColony/colonyDapp/pull/3527/files#diff-692e723973e10e96eea291e621374752d4483680460c120175c41afed277eef3
I don't want to copy the same code here.
2. It is similar to the first point. Claim delay [PR](https://github.com/JoinColony/colonyDapp/pull/3527) isn't merged yet to the [advanced-expenditures](https://github.com/JoinColony/colonyDapp/tree/feature/advanced-expenditures) branch, so delay doesn't work yet.
3. and 4. I've added fixes. Thank you for pointing this out.
5. I couldn't find tooltip for this dot in the desing. @arrenv should tooltip be here?  If so, can you please tell me what should be displayed here?

Julianahlk commented:
> 1. I've fixed 'Completed' button color in this PR: https://github.com/JoinColony/colonyDapp/pull/3527/files#diff-692e723973e10e96eea291e621374752d4483680460c120175c41afed277eef3
>    I don't want to copy the same code here.
> 2. It is similar to the first point. Claim delay [PR](https://github.com/JoinColony/colonyDapp/pull/3527) isn't merged yet to the [advanced-expenditures](https://github.com/JoinColony/colonyDapp/tree/feature/advanced-expenditures) branch, so delay doesn't work yet.
> 3. and 4. I've added fixes. Thank you for pointing this out.
> 4. I couldn't find tooltip for this dot in the desing. @arrenv should tooltip be here?  If so, can you please tell me what should be displayed here?

1. I see the changes now. Thank you!
2. Got it!
3. Happy to help!

chinins commented on src/modules/dashboard/components/ExpenditurePage/Stages/LockedStages.tsx:
```diff
@@ -0,0 +1,81 @@
+  },
+});
+
+const displayName = 'dashboard.ExpenditurePage.LockedStages';
```
missing Stages

chinins commented on src/modules/dashboard/components/ExpenditurePage/Stages/LockedStages.tsx:
```diff
@@ -0,0 +1,81 @@
+
+const MSG = defineMessages({
+  motion: {
+    id: 'dashboard.ExpenditurePage.Stages.motion',
```
missing LockedStages

chinins commented on src/modules/dashboard/components/ExpenditurePage/Stages/Stages.tsx:
```diff
@@ -115,8 +109,46 @@ const Stages = ({ states, activeStateId }: Props) => {
+              <FormattedMessage {...label} />
+            ) : (
+              label
+            )}{' '}
```
Needs to be combined in a translation string and used with FormattedMessage. Not good to have spaces like this if possible

chinins reviewed (changes requested):
Nice work - a lot of it :)

This PR is very large and difficult to review. PRs should be smaller. If you fee that it's getting large (like this one), please, talk to Arren to split the issue.

In general, I would prefer for you to split the code in more digestible parts (make more subcomponents). I have commented it in specific places, but this is an overall guidance. In many cases the files/components are too big and complicated, they need to be split up.

I have a warning in console. Please, fix it.
<img width="481" alt="Screenshot 2022-08-04 at 16 49 58" src="https://user-images.githubusercontent.com/34057551/182865385-5be011f8-f958-4cba-9401-8f18340c15af.png">

The token & amount are of different font/alignment (in figma they are the same)
<img width="252" alt="Screenshot 2022-08-04 at 16 51 57" src="https://user-images.githubusercontent.com/34057551/182865508-bfd0d341-480d-47d6-97f7-99d3d22bb3a2.png">

Please, ping me once you finish with the updates - I'll make sure to review with priority (while it's also more fresh in my mind).

arrenv commented on src/modules/dashboard/components/Dialogs/EditExpenditureDialog/EditExpenditureDialogForm.tsx:
```diff
@@ -0,0 +1,320 @@
+  },
+  descriptionText: {
+    id: `dashboard.EditExpenditureDialog.EditExpenditureDialogForm.descriptionText`,
+    defaultMessage: `Payment is currently at the locked stage.
```
Please change this text to be: "This Payment is currently Locked. Either a Motion, or a member with the Arbitration permission are required to make changes. "

arrenv commented:
@agnieszkajarosikloj Regarding the Tooltip, which I seemed to have missed the notification for. Please use this copy:

> There is an active Motion to change this payment.

As of the design here- https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=5675%3A322900

arrenv reviewed (changes requested):
Thank you @agnieszkajarosikloj, this is certainly a big one. Some feedback:

1. This padding is uneven in the modal

![37d84219-a4dc-4837-b445-dc1582c53cdf](https://user-images.githubusercontent.com/33682027/183439552-44e76f7d-3abf-40ec-9a14-df6070c53ba3.png)

2. These are inconsistent, please use this format for both: `1: @username, 100 TKN, 1 hours`

![download](https://user-images.githubusercontent.com/33682027/183442418-3909e103-1fa8-4346-bf8c-04563b4de292.png)

![download (1)](https://user-images.githubusercontent.com/33682027/183442459-1d0af2e2-bcc1-41f5-aa32-bf23fb55f007.png)


3. I added a new recipient and then discarded it in the modal and I received the following message, also nothing happened when click the "Discard" button on the 1st recipient.

![863219b5-6975-416e-b4df-0a53fe95c744](https://user-images.githubusercontent.com/33682027/183443128-e180330a-8662-4e96-9b9f-509341ceb22e.png)

4. Could we please have consistent fonts for each of these:

![download (2)](https://user-images.githubusercontent.com/33682027/183443982-788e4942-addf-43a9-a3c5-554f47838a50.png)

Please have them all match this font:

```
font-size: var(--size-normal);
font-weight: 700;
 ```

5. Please update this message in yellow to match the design:

![8dbda23a-06b2-4d1d-b74b-554dfd67c3bd](https://user-images.githubusercontent.com/33682027/183448115-0f0a5a9b-8efa-416a-be35-a60620af7ad9.png)

![Screenshot 2022-08-08 165449](https://user-images.githubusercontent.com/33682027/183448166-30db0bf1-48a2-45a7-99c5-457b2094fd09.png)

chinins reviewed (changes requested):
Much better - easier to read the code and understand all the parts.
On top of my direct comments, I've noticed the following:

1. The css warning is still there
<img width="1204" alt="Screenshot 2022-08-08 at 18 43 47" src="https://user-images.githubusercontent.com/34057551/183461181-cd667503-f391-4d4b-be97-fd62d13c4cb0.png">



2. There is a `key` warning in the edit dialog - I guess because you use id as `key` and it's the same users that are being changed. Also claim delay for the second user shouldn't be there as it didn't change/wasn't added.

<img width="1363" alt="Screenshot 2022-08-08 at 18 49 00" src="https://user-images.githubusercontent.com/34057551/183461008-6ef2cedc-b812-4213-a140-10ca020b2153.png">

3. I encounter a bug (similar to Arren's) when I click on discard of the second user - just claim delay disappears and all values switched to be the same for both items
<img width="639" alt="Screenshot 2022-08-08 at 18 51 07" src="https://user-images.githubusercontent.com/34057551/183461233-9e0f905c-dda7-4549-9c5f-f7ca09eb97da.png">

This is what happens when I click discard second time 
<img width="719" alt="Screenshot 2022-08-08 at 18 51 23" src="https://user-images.githubusercontent.com/34057551/183461414-1ab50d87-29c6-4c48-826d-7aab924f2252.png">

Julianahlk commented:
> @Julianahlk thank you for comments.

> 1. I've fixed 'Completed' button color in this PR: https://github.com/JoinColony/colonyDapp/pull/3527/files#diff-692e723973e10e96eea291e621374752d4483680460c120175c41afed277eef3
>    I don't want to copy the same code here.
> 2. It is similar to the first point. Claim delay [PR](https://github.com/JoinColony/colonyDapp/pull/3527) isn't merged yet to the [advanced-expenditures](https://github.com/JoinColony/colonyDapp/tree/feature/advanced-expenditures) branch, so delay doesn't work yet.
> 3. and 4. I've added fixes. Thank you for pointing this out.
> 4. I couldn't find tooltip for this dot in the desing. @arrenv should tooltip be here?  If so, can you please tell me what should be displayed here?

I see the changes you made. Looking good!
<img width="1728" alt="Screen Shot 2022-08-11 at 10 01 23 PM" src="https://user-images.githubusercontent.com/71563622/184234058-beeaaf6c-68de-4492-83d0-d095f831e584.png">
<img width="494" alt="Screen Shot 2022-08-11 at 10 06 40 PM" src="https://user-images.githubusercontent.com/71563622/184234077-dfcec257-f251-4a49-96d7-a35e3a236ff4.png">

A few things:
- Can we give the orange tag below a tiny bit more of top and bottom margin space?
<img width="431" alt="Screen Shot 2022-08-11 at 9 52 13 PM" src="https://user-images.githubusercontent.com/71563622/184234404-2acd0999-532a-4991-b002-b8df5aa60138.png">

When I pick an username that does not exist I get the following:
<img width="1699" alt="Screen Shot 2022-08-11 at 10 21 04 PM" src="https://user-images.githubusercontent.com/71563622/184235230-53ae568c-16bc-423f-87f3-f521f9ae87f6.png">

Julianahlk commented:
Hey @agnieszkajarosikloj I am still seeing the following warnings:
<img width="1447" alt="Screen Shot 2022-08-16 at 12 08 40 AM" src="https://user-images.githubusercontent.com/71563622/184727425-148dd847-40b8-48b1-9b01-8e6bfec76390.png">

<img width="1396" alt="Screen Shot 2022-08-16 at 12 09 43 AM" src="https://user-images.githubusercontent.com/71563622/184728354-56908f31-229b-4055-aa0f-530cc088ca2e.png">

But things are looking great! Good work 💯🙌✨
<img width="1382" alt="Screen Shot 2022-08-16 at 12 18 17 AM" src="https://user-images.githubusercontent.com/71563622/184728482-f32376fc-43c2-417d-abb8-6abd91ba02f1.png">

arrenv reviewed (changes requested):
Thank you @agnieszkajarosikloj, I am still seeing these issues.

2. These are inconsistent, please use this format for both: `1: @username, 100 TKN, 1 hours`

![download](https://user-images.githubusercontent.com/33682027/183442418-3909e103-1fa8-4346-bf8c-04563b4de292.png)

![download (1)](https://user-images.githubusercontent.com/33682027/183442459-1d0af2e2-bcc1-41f5-aa32-bf23fb55f007.png)

4. Could we please have consistent fonts for each of these:

![download (2)](https://user-images.githubusercontent.com/33682027/183443982-788e4942-addf-43a9-a3c5-554f47838a50.png)

Please have them all match this font:

```
font-size: var(--size-normal);
font-weight: 700;
 ```

I also received the same unique key prop error as @Julianahlk, however, I don't seem to be able to replicate it too easily.

arrenv commented on src/modules/dashboard/components/ExpenditurePage/Payments/LockedPayments.tsx:
```diff
@@ -69,75 +77,91 @@ const LockedPayments = ({
+          );
 
           return (
             <div className={styles.singleRecipient} key={recipient.id}>
```
This is a key that is causing errors for me when I use the same recipient for multiple payments. I think we could use the count as well as the recipient.id i.e. address as the key. For example `0xaEsd0...wE73_1`.

![31620875-2a03860425fd52af46b745d2222db8e3](https://user-images.githubusercontent.com/33682027/185993354-a40b1ea8-5eca-44fd-b406-526ef725ed6e.png)

arrenv commented on src/modules/dashboard/components/ExpenditurePage/Payments/LockedPayments.tsx:
```diff
@@ -69,75 +77,91 @@ const LockedPayments = ({
+      <div className={styles.recipientContainer}>
+        {pendingCreations?.map((newRecipient, index) => {
+          return (
+            <div className={styles.singleRecipient} key={newRecipient.id}>
```
Same here, please use the count as well as the recipient.id i.e. address as the key. For example `0xaEsd0...wE73_1`

arrenv commented on src/modules/dashboard/components/Dialogs/EditExpenditureDialog/ChnagedRecipients/ChangedRecipients.tsx:
```diff
@@ -0,0 +1,212 @@
+                                ({ amount, token }) =>
+                                  token &&
+                                  amount && (
+                                    <div key={token.id}>
```
This seems to be causing an issue when the same token is added twice.

arrenv commented on src/modules/dashboard/components/ExpenditurePage/Payments/RecipientHeader/RecipientHeader.tsx:
```diff
@@ -0,0 +1,139 @@
+                    ({ amount, token }) =>
+                      token &&
+                      amount && (
+                        <div className={styles.tokenWrapper} key={token.id}>
```
Would this cause key issues if you add the same token twice?

arrenv reviewed (changes requested):
@agnieszkajarosikloj Thank you, the updates look good. I think I identified the unique "key" issues, please check my comments.

arrenv reviewed (approved):
Lots of work on this, looking good to me, thank you @agnieszkajarosikloj

chinins reviewed (approved):
That is a lot of work - well done!

Julianahlk reviewed (approved):
Great work @agnieszkajarosikloj 🙌
